Sources/TuistGenerator/Writers/XcodeProjWriter.swift

```diff
@@ -0,0 +1,99 @@
+import Basic
+import Foundation
+import TuistSupport
+import XcodeProj
+
+public protocol XcodeProjWriting {
```

Comment:
Does this have to be public?

Reply:
It does, as the descriptor generator returns descriptors, the caller can choose to modify those descriptors ahead of writing them to disk.